llama_cpp/llama.py

```diff
@@ -1369,11 +1370,24 @@ def _create_completion(
         logits_processor: Optional[LogitsProcessorList] = None,
         grammar: Optional[LlamaGrammar] = None,
         logit_bias: Optional[Dict[str, float]] = None,
+        hf_tokenizer: Optional[Any] = None,
```

Comment:
Can we introduce a `typing.Protocol` for a tokenizer and move this up to the `Llama.__init__` as an optional override to the default tokenizer.